app/src/main/java/com/sdp13epfl2021/projmag/activities/CommentsActivity.kt

```diff
@@ -1,24 +1,87 @@
 package com.sdp13epfl2021.projmag.activities
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.PendingIntent.getActivity
+import android.content.Context
 import android.os.Bundle
 import android.util.Log
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.ImageButton
+import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.RecyclerView
 import com.sdp13epfl2021.projmag.R
 import com.sdp13epfl2021.projmag.adapter.MessageListAdapter
+import com.sdp13epfl2021.projmag.database.interfaces.CommentsDatabase
+import com.sdp13epfl2021.projmag.database.interfaces.ProjectId
+import com.sdp13epfl2021.projmag.database.interfaces.UserdataDatabase
 import com.sdp13epfl2021.projmag.model.*
+import dagger.hilt.android.AndroidEntryPoint
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Named
 
+@AndroidEntryPoint
 class CommentsActivity : AppCompatActivity() {
     private lateinit var recyclerView: RecyclerView
     private lateinit var messageAdapter: MessageListAdapter
+    private lateinit var comments : List<Message>
+
+
+
+    @Inject
+    lateinit var  commentsDB : CommentsDatabase
+
+    @Inject
+    lateinit var userDB: UserdataDatabase
+
+    @Inject
+    @Named("currentUserId")
+    lateinit var userId: String
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_comments)
-
-        val comments = MessagesDatasource().loadMessages()
+        //get the users profileId
+        val profileId : String = userId
+        // send a message
+        // get the project id
+        val projectId = intent.extras?.getString("projectId")
```

Comment:
We should try to avoid using string literal for name of extra.
```suggestion
        val projectId: String? = intent.getStringExtra(MainActivity.projectIdString)
```